File: 114-flatten-binary-tree-to-linked-list/flatten-binary-tree-to-linked-list.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private TreeNode prev = null;

    private void flattenT(TreeNode root){
        if(root == null){
            return;
        }
        flattenT(root.right);
        flattenT(root.left);

        root.right = prev;
        root.left = null;
        prev = root;
    }
    public void flatten(TreeNode root) {
        flattenT(root);
    }
}

File: 114-flatten-binary-tree-to-linked-list/flatten-binary-tree-to-linked-list.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def flatten(self, root: Optional[TreeNode]) -> None:

        self.prev = None

        def traverse(root):
            if not root:
                return
            
            traverse(root.right)
            traverse(root.left)

            root.right = self.prev
            root.left = None
            self.prev = root

        traverse(root)    